### PR TITLE
feat(NODE-6008): deprecate CloseOptions interface

### PR DESCRIPTION
**Pull request by W-A-James**
#### What is changing?
* Deprecated `CloseOptions` interface

##### Is there new documentation needed for these changes?
No

#### What is the motivation for this change?
NODE-6008

### Deprecated unused `CloseOptions` interface

The `CloseOptions` interface was unintentionally made public and was only intended for use in the driver's internals. Due to recent refactoring (NODE-5915), this interface is no longer used in the driver. Since it was marked public, out of an abundance of caution we will not be removing it outside of a major version, but we have deprecated it and will be removing it in the next major version.

### Double check the following

- [X] Ran `npm run check:lint` script
- [X] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [X] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

**Comment by baileympearson** [comment]
@W-A-James Is the base branch for this PR correct?  (CI didn't run because it's not pointing to `main`)

**Review by aditi-khare-mongoDB — COMMENTED** [review]
code changes are minimal and LGTM, but do we know what's going on with the Ci alias?

**Comment by baileympearson** [comment]
evergreen retry

**Comment by W-A-James** [comment]
> @W-A-James Is the base branch for this PR correct? (CI didn't run because it's not pointing to `main`)

I didn't want to mark this as internal until we actually stop using it (once NODE-5915 is merged), so I pointed it at the branch with the changes that remove its usages.

**Review by aditi-khare-mongoDB — DISMISSED** [review]

File: src/cmap/connection_pool.ts [hunk 1]
```diff
@@ -118,7 +118,9 @@ export const PoolState = Object.freeze({
   closed: 'closed'
 } as const);
 
-/** @public */
+/** @public
+ * @deprecated This interface is deprecated and will be removed in a future release as it is not used
+ * in the driver */
```
Comment (nbbeeken) [inline src/cmap/connection_pool.ts:123]:
```suggestion
/** 
 * @public
 * @deprecated This interface is deprecated and will be removed in a future release as it is not used
 * in the driver 
 */
```

**Review by nbbeeken — CHANGES_REQUESTED** [review]

**Review by aditi-khare-mongoDB — APPROVED** [review]

**Review by nbbeeken — DISMISSED** [review]

**Review by aditi-khare-mongoDB — APPROVED** [review]

**Pull request merged**